src/main/scala/com/mwz/sonar/scala/qualityprofiles/RecommendedQualityProfile.scala

```diff
@@ -37,34 +34,14 @@ final class RecommendedQualityProfile extends BuiltInQualityProfilesDefinition {
     val profile: NewBuiltInQualityProfile =
       context.createBuiltInQualityProfile(ScalastyleScapegoatQualityProfile.ProfileName, Scala.LanguageKey)
 
-    // Enable Scalastyle rules - excluding those which have been blacklisted and templates.
-    ScalastyleInspections.AllInspections
-      .filterNot { i =>
-        RecommendedQualityProfile.Blacklist.contains(i.id) ||
-        i.params.nonEmpty
-      }
-      .foreach { inspection =>
-        RecommendedQualityProfile.processRule(
-          profile,
-          ScalastyleRulesRepository.RepositoryKey,
-          inspection.clazz,
-          inspection.id
-        )
-      }
+    // Enable Scalastyle rules excluding blacklisted rules and templates.
+    // Overrides the default severity and parameter values.
+    ScalastyleQualityProfile.activateWithOverrides(profile, RecommendedQualityProfile.RuleOverrides)
 
     // TODO: Scalastyle template instances.
 
-    // Enable Scapegoat rules - excluding those which have been blacklisted.
-    ScapegoatInspections.AllInspections
-      .filterNot(inspection => RecommendedQualityProfile.Blacklist.contains(inspection.id))
-      .foreach { inspection =>
-        RecommendedQualityProfile.processRule(
-          profile,
-          ScapegoatRulesRepository.RepositoryKey,
-          inspection.id,
-          inspection.id
-        )
-      }
+    // Enable Scapegoat rules excluding blacklisted rules. Overrides the severity.
+    ScapegoatQualityProfile.activateWithOverrides(profile, RecommendedQualityProfile.RuleOverrides)
 
     // Ensure this is not the default profile.
     profile.setDefault(false)
```

Comment:
I just noticed, this should be the default profile, right?
Or what do you think?, I mean for a fresh install this should be best option.
And, I'm sure that if one creates another profile which he/she wants to be the default it can be configured from the SonarQube web interface, and Sonar will override the settings - or at least I hope so.

Reply:
yeah I guess so - I'll try it out and see what happens